src/Implement/UnitTestFileOperator.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <arpa/inet.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <memory>
+#include <netinet/in.h>
+#include <thread>
+#include <unistd.h>
+
+#include "OperateData.h"
+#include "FileOperator.h"
+
+uint16_t test_port = 9000;
+std::string temp_response = "";
+
+void TestClient() {
+    int sockfd = 0;
+    sockfd = socket(AF_INET, SOCK_STREAM, 0);
+    struct sockaddr_in info;
+    bzero(&info, sizeof(info));
+    info.sin_family = PF_INET;
+    info.sin_addr.s_addr = inet_addr("127.0.0.1");
+    info.sin_port = htons(test_port);
+    int err = connect(sockfd, (struct sockaddr *)&info, sizeof(info));
+
+    char buffer[2048] = {};
+    size_t n_recv = recv(sockfd, buffer, sizeof(buffer), 0);
+
+    temp_response = buffer;
+
+    close(sockfd);
+}
+
+// test with a sudo string data
+TEST(RESPONSEHANDLER_TEST, Basic) {
```

Comment:
Here you don't have the issue of not being able to mock a socket, so you can create a MockResponseHandler and expect the result in MockResponseHandler instead of going through the socket.